src/co_pdo.c

```diff
@@ -594,6 +594,50 @@ void co_pdo_trigger (co_net_t * net)
    }
 }
 
+void co_pdo_trigger_with_obj (co_net_t * net, uint16_t index, uint8_t subindex)
+{
+   unsigned int ix;
+   uint8_t n;
+   const co_obj_t * obj;
+   const co_entry_t * entry;
+
+   /* Find mapped object */
+   obj = co_obj_find (net, index);
+   if (obj == NULL)
+      return;
+
+   /* Find mapped entry */
+   entry = co_entry_find (net, obj, subindex);
+   if (entry == NULL)
+      return;
+
+   /* Check that object is mappable */
+   if ((entry->flags & OD_TPDO) == 0)
+      return;
+
+   /* Transmit event-driven TPDOs, queue acyclic TPDOs */
+   for (ix = 0; ix < MAX_TX_PDO; ix++)
+   {
+      co_pdo_t * pdo = &net->pdo_tx[ix];
+
```

Comment:
```suggestion
      if (pdo->cobid & CO_COBID_INVALID)
         continue;
         
```

Reply:
There's probably no need to search PDO:s that are invalid?

Reply:
True, and I guess that could even cause disabled PDOs to be sent.